observatory-platform/observatory/platform/sql/upsert_records.sql.jinja2

```diff
@@ -14,7 +14,15 @@
 
 MERGE `{{ main_table_id }}` main_table
 USING `{{ upsert_table_id }}` upsert_table
-ON main_table.{{ primary_key }} = upsert_table.{{ primary_key }}
+{% if keys|length == 1 %}
```

Comment:
The jinja template could be simplified a little bit, I've also remove the newlines in the output (which were already there) with the minus signs:

```jinja2
MERGE `{{ main_table_id }}` main_table
USING `{{ upsert_table_id }}` upsert_table
ON (
  {%- for key in keys %}
  main_table.{{ key }}=upsert_table.{{ key }}{% if not loop.last %} AND{% endif %}
  {%- endfor %}
)
WHEN MATCHED THEN
  UPDATE SET
    {%- for col in columns %}
    main_table.{{ col }}=upsert_table.{{ col }}{% if not loop.last %},{% endif %}
    {%- endfor %}
WHEN NOT MATCHED THEN
  INSERT ROW
```

Then the output would be:
```sql
MERGE `my-project.bq_tests_c9b656431a145bdb8b9c22130bbf81f1.presidents` main_table
USING `my-project.bq_tests_c9b656431a145bdb8b9c22130bbf81f1.presidents_upserts` upsert_table
ON (
  main_table.name=upsert_table.name AND
  main_table.death_age=upsert_table.death_age
)
WHEN MATCHED THEN
  UPDATE SET
    main_table.name=upsert_table.name,
    main_table.death_age=upsert_table.death_age,
    main_table.date=upsert_table.date
WHEN NOT MATCHED THEN
  INSERT ROW
```